ui/admin/app/controllers/account/change-password.js

```diff
@@ -5,9 +5,41 @@
 
 import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
 
 export default class AccountChangePasswordController extends Controller {
   // =services
 
   @service session;
+  @service router;
+
+  // =actions
+
+  /**
+   * Changes the password on the account and then redirects to the index route.
+   * @param {AccountModel} account
+   * @param {string} currentPassword
+   * @param {string} newPassword
+   */
+  @action
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.save-success')
+  async changePassword(account, currentPassword, newPassword) {
+    await account.changePassword(currentPassword, newPassword);
+    // Transition to index after success.  We ignore errors in the transition,
+    // since the index route is responsible for additional
+    // redirects depending on the authentication context.  These additional
+    // redirects may abort this initial transition, resulting in a
+    // TransitionAborted error, which we do not want to show to the user.
+    await this.router.replaceWith('index').catch(() => {});
```

Comment:
I know this is just personal taste but I'm not the biggest fan of mixing both `await` and using `.catch`. I think if we're awaiting we should just handle this in a normal try catch block but I'm not sure what other's opinions are. 

In this simple use case, it doesn't end up mattering since we just swallow the error and I guess it's fine if this is the only time we use it, but I'm personally not the biggest fan because in general you now have to think about it in both a standard synchronous way using `await` but still being mindful of the asynchronous `.catch` callback. This [article](https://www.robinwieruch.de/javascript-async-await-without-try-catch/) might explain it a bit better.

I think it's fine if we say we only use it in this type of scenario but I also don't think it's a big deal to be consistent and just use try catch?

Reply:
Thank you for the resource and detailed explanation, i agree with updating this. Also using try catch IMO is easier to understand.